### PR TITLE
specgen: do not set OOMScoreAdj by default

### PR DESCRIPTION
**Pull request by giuseppe**
do not force a value of OOMScoreAdj=0 if it is wasn't specified by the
user.

Closes: https://github.com/containers/podman/issues/13731

Signed-off-by: Giuseppe Scrivano <gscrivan@redhat.com>

### CODE REVIEW

File: hack/libsubid_tag.sh [hunk 1]
```diff
@@ -24,6 +24,3 @@ int main() {
 	return 0;
 }
 EOF
-if test $? -eq 0 ; then
-	echo libsubid
-fi
```
Comment (vrothberg) [inline hack/libsubid_tag.sh:29]: Could you move that into a separate commit?  We probably need to cherry-pick the OOM fix.

File: test/e2e/run_test.go [hunk 2]
```diff
@@ -609,6 +609,13 @@ USER bin`, BB)
 		session.WaitWithDefaultTimeout()
 		Expect(session).Should(Exit(0))
 		Expect(session.OutputToString()).To(Equal("111"))
+
+		currentOOMScoreAdj, err := ioutil.ReadFile("/proc/self/oom_score_adj")
```
Comment (vrothberg) [inline test/e2e/run_test.go:613]: Could we turn this into a system test?  Just to be extra sure we catch any potential regression in RHEL (gating).

File: cmd/podman/common/create.go [hunk 3]
```diff
@@ -348,8 +348,7 @@ func DefineCreateFlags(cmd *cobra.Command, cf *entities.ContainerCreateOptions,
 		)
 
 		oomScoreAdjFlagName := "oom-score-adj"
-		createFlags.IntVar(
-			&cf.OOMScoreAdj,
+		createFlags.Int64(
```
Comment (Luap99) [inline cmd/podman/common/create.go:351]: Flag should be `Int` and not `Int64` since you cast later to an int again.

File: cmd/podman/containers/create.go [hunk 4]
```diff
@@ -238,6 +238,15 @@ func CreateInit(c *cobra.Command, vals entities.ContainerCreateOptions, isInfra
 			vals.GroupAdd = groups
 		}
 
+		if c.Flags().Changed("oom-score-adj") {
+			val := c.Flag("oom-score-adj").Value.String()
```
Comment (Luap99) [inline cmd/podman/containers/create.go:242]: use `c.Flags().GetInt("oom-score-adj")` instead

File: hack/libsubid_tag.sh (hunk 1 above)
Comment (Luap99) [inline hack/libsubid_tag.sh:29]: This doesn't look related to this PR/commit

**Review by vrothberg — COMMENTED** [review]
Thanks a lot, @giuseppe!

Which API are you referring to re: the breaking change?  I fail to find the break in the remote API which we should keep stable.

File: hack/libsubid_tag.sh (hunk 1 above)
Comment (giuseppe) [reply hack/libsubid_tag.sh]: no that is some debug change that should not be committed

**Comment by vrothberg** [comment]
@mheon PTAL, a candidate for another v4.0 backport.

File: test/system/030-run.bats [hunk 5]
```diff
@@ -815,4 +815,11 @@ EOF
     run_podman run --uidmap 0:10001:10002 --rm --hostname ${HOST} $IMAGE grep ${HOST} /etc/hosts
     is "${lines[0]}" ".*${HOST}.*"
 }
+
+@test "podman run doesn't override oom-score-adj" {
+    current_oom_score_adj=$(cat /proc/self/oom_score_adj)
+    run_podman '?' run --rm $IMAGE cat /proc/self/oom_score_adj
```
Comment (Luap99) [inline test/system/030-run.bats:821]:
```suggestion
    run_podman run --rm $IMAGE cat /proc/self/oom_score_adj
```
We should make sure exit code is 0 or is there a special reason to not check for that?

File: test/system/030-run.bats [hunk 6]
```diff
@@ -815,4 +815,11 @@ EOF
     run_podman run --uidmap 0:10001:10002 --rm --hostname ${HOST} $IMAGE grep ${HOST} /etc/hosts
     is "${lines[0]}" ".*${HOST}.*"
 }
+
+@test "podman run doesn't override oom-score-adj" {
+    current_oom_score_adj=$(cat /proc/self/oom_score_adj)
+    run_podman run --rm $IMAGE cat /proc/self/oom_score_adj
+    is "$output" "$current_oom_score_adj" "different oom_score_adj in the container"
+
```
Comment (vrothberg) [inline test/system/030-run.bats:823]: Nit: trailing line.

**Review by vrothberg — COMMENTED** [review]
LGTM, thanks!

File: cmd/podman/common/create_opts.go [hunk 7]
```diff
@@ -286,7 +286,7 @@ func ContainerCreateToContainerCLIOpts(cc handlers.CreateContainerConfig, rtc *c
 		LogDriver:         cc.HostConfig.LogConfig.Type,
 		LogOptions:        stringMaptoArray(cc.HostConfig.LogConfig.Config),
 		Name:              cc.Name,
-		OOMScoreAdj:       cc.HostConfig.OomScoreAdj,
```
Comment (mheon) [inline cmd/podman/common/create_opts.go:289]: This looks suspicious - this is for the Docker-compat API, and I don't see any other place it could be set for said API, so we're basically just ignoring OOMScoreAdj for the Docker-compat create endpoint now?

Comment (giuseppe) [reply cmd/podman/common/create_opts.go:289]: I guess it is fine in this case to override the default as we cannot say whether it is set to 0 or undefined.

I've reverted this line.
